youtube_dl/downloader/external.py

```diff
@@ -136,6 +141,66 @@ def _make_cmd(self, tmpfilename, info_dict):
             cmd += ['%s:%s' % (key, val)]
         return cmd
 
+
+class FFmpegFD(ExternalFD):
+    @classmethod
+    def supports(cls, info_dict):
+        return info_dict['protocol'] in ('http', 'https', 'ftp', 'ftps', 'm3u8', 'rtsp', 'rtmp', 'mms')
+
+    def _call_downloader(self, tmpfilename, info_dict):
+        url = info_dict['url']
+        ffpp = FFmpegPostProcessor(downloader=self)
+        ffpp.check_version()
+
+        args = [ffpp.executable, '-y']
+
+        start_time = info_dict.get('start_time', 0)
+        if start_time:
+            args += ['-ss', compat_str(start_time)]
+        end_time = info_dict.get('end_time')
+        if end_time:
+            args += ['-t', compat_str(end_time - start_time)]
```

Comment:
> start_time may be None at this point.

fixed with f34294fa0c0097cea7f6388d5d691d5a54950491.
> end_time could be < than start_time producing negative duration.

this should be handled in general may be in YoutubeDL.process_ie_result.